tests/acceptance/guid-file/registration-file-detail-test.ts

```diff
@@ -294,4 +295,86 @@ module('Acceptance | guid file | registration files', hooks => {
+        await fillIn('[data-test-title-field] > div > textarea', 'A New Title');
+        // Update description
+        await fillIn('[data-test-description-field] > div > textarea', 'A New Description');
+        // Update resource type
+        await selectChoose('[data-test-select-resource-type]', 'InteractiveResource');
+        // Update resource language
+        await selectChoose('[data-test-select-resource-language]', 'English');
+        // Save changes
+        await click('[data-test-save-metadata-button]');
+
+        assert.dom('[data-test-file-title]').hasText('A New Title',
+            'File metadata file title is properly updated.');
+        assert.dom('[data-test-file-description]').hasText('A New Description',
+            'File metadata file description is properly updated.');
+        assert.dom('[data-test-file-resource-type]').hasText('InteractiveResource',
+            'File metadata resource type is properly updated.');
+        assert.dom('[data-test-file-language]').hasText('English',
+            'File metadata resource language is properly updated.');
+    });
+
+    test('View-only section does not update', async function(this: ThisTestContext, assert) {
+        await visit(`/--file/${this.file.id}`);
+
+        await click('[data-test-edit-metadata-button]');
+        // Update title
+        await fillIn('[data-test-title-field] > div > textarea', 'A New Title');
+        // Update description
+        await fillIn('[data-test-description-field] > div > textarea', 'A New Description');
+        // Update resource type
+        await selectChoose('[data-test-select-resource-type]', 'InteractiveResource');
+        // Update resource language
+        await selectChoose('[data-test-select-resource-language]', 'Latin');
+        // Cancel changes
+        await click('[data-test-cancel-metadata-button]');
+
+        assert.dom('[data-test-file-title]').doesNotHaveTextContaining('A New Title',
+            'Cancel metadata edit button properly working for title.');
+        assert.dom('[data-test-file-description]').doesNotHaveTextContaining('A New Description',
+            'Cancel metadata edit button properly working for description.');
+        assert.dom('[data-test-file-resource-type]').doesNotHaveTextContaining('InteractiveResource',
+            'Cancel metadata edit button properly working for resource type.');
+        assert.dom('[data-test-file-language]').doesNotHaveTextContaining('Latin',
+            'Cancel metadata edit button properly working for resource language.');
+    });
+
+    test('Error and canceling edit', async function(this: ThisTestContext, assert) {
+        await visit(`/--file/${this.file.id}`);
+
+        await click('[data-test-edit-metadata-button]');
+        await fillIn('[data-test-title-field] > div > textarea', 'A New Title');
+        await server.get(`/--file/${this.file.id}`, {}, 500);
+        await server.get(`/--file/${this.file.id}`, {}, 200);
```

Comment:
For this test, if we are trying to make it error when trying to update the metadata, you'll want to check which endpoint is being hit when we save by checking the console, and override that endpoint with the response we want it to send back.

This test has a couple of examples of endpoints being overridden to simulate a server error that may help: https://github.com/CenterForOpenScience/ember-osf-web/blob/develop/tests/engines/registries/acceptance/edit-revision/revision-test.ts#L409-L415

Reply:
Thank you, that sounds good. I'll take a look.